tests/Doctrine/Tests/DBAL/Functional/Driver/OCI8/StatementTest.php

```diff
@@ -39,17 +41,53 @@ public function testQueryConversion(string $query, array $params, array $expecte
         );
     }
 
+    /**
+     * Low-level approach to working with parameter binding
+     *
+     * @param mixed[] $params
+     * @param mixed[] $expected
+     *
+     * @dataProvider queryConversionProvider
+     */
+    public function testStatementBindParameters(string $query, array $params, array $expected) : void
+    {
+        $stmt         = $this->connection->prepare($query);
+        $hasZeroIndex = array_key_exists(0, $params);
+
+        foreach ($params as $key => $val) {
+            if ($hasZeroIndex && is_int($key)) {
```

Comment:
Does this test cover any code path not covered by the data item added to `queryConversionProvider()`. This test seems to duplicate the code being tested and has conditional operators which usually identify a poorly designed test.

Reply:
ok, changed it.  

The `testQueryConversion` test does not cover all use cases because it calls the `executeQuery` method on the `connection` object. In this case, the named parameters are transformed into positional ones. As a result, test cases with named parameters are skipped